libvast/vast/type.hpp

```diff
@@ -671,6 +674,10 @@ class list_type final : public stateful_type_base {
     // nop
   }
 
+  /// Returns the underlying FlatBuffers table representation.
+  /// @note Include `vast/fbs/type.hpp` to be able to use this function.
+  [[nodiscard]] const fbs::Type& table() const noexcept;
```

Comment:
To me, this should rather be a free function in the namespace `fbs`, e.g., `template <class T> const T& fbs::get();`. It's an implementation detail and not part of the primary interface the class `type` provides to the user.

Reply:
I disagree. `get` is a terrible name, and it is not an implementation detail but rather part of the APIs of `type` and all complex concrete types, because you might want to fall back to using the underlying FlatBuffers API directly.

Reply:
> but rather part of the APIs of `type`

Calling `.table()` doesn't do any meaningful type operation, it's purely exposing a different representation.

> because you might want to fall back to using the underlying FlatBuffers API directly

We have the same for `pack` and `unpack`. These are flatbuffers-specific CPOs. Here, exposing an implementation detail, which a lot of *different* flatbuffers-backed types do, should be a free function. Whether you call it `get` or something else is irrelevant for this discussion, e.g., `flatbuffer(x)` or `table(x)` are probably nicer.

Reply:
It'd still be majorly confusing, because for a given `record_type x`, `table(rt)` would return the same type as `table(type{rt})`, but with different assumptions that can be made about the resulting type. It's an operation on the type that for concrete complex types resolves some layers transparently.

It's also not a CPO. We can introduce a CPO when we have generic code that calls the CPO via ADL or similar means, but in this case it really isn't that—there's nothing you can do on the result of this function generically.

Reply:
I think #1972 offers a better solution to this. Once we no longer have the chunk we can make this all a bit simpler and remove the `table()` function entirely. It's a future refactoring, though, and I would like to keep it out of this PR, so I am resolving this for now.